### ISSUE TITLE
PHP executable not found. You need PHP 7 installed and in your PATH

### ISSUE THREAD

**Issue by FAST-JE** [Microsoft/vscode#13356]
- VSCode Version: 1.5.3
- OS Version: Windows 10 x64

Hi, Humans.
Steps to Reproduce:

1. I installed VS code and then the problem started.
2. I opened VS and then I installed ftp-simple extension.
3. I opened project next I opened file with php extension.
4. Then I saw this info 

> Cannot validate the php file. The php program was not found. Use the 'php.validate.executablePath' setting to configure the location of 'php'

I said okey, and I opened 

> FIle > Preferences > User Settings > Settings.json 

I added this line 

> // Points to the php executable.
    "php.validate.executablePath":"C:\\\OpenServer\\\modules\\\php\\\PHP-7-x64\\\php.exe",

The problem is gone.
5. But then I saw new error 

> PHP executable not found. You need PHP 7 installed and in your PATH

How to solve this error I don't know. Please, help me.

**Comment by dkimery256** [comment]
I am having the same problem. I wanted to use the IntelliSense extension for VS Code editor and when I made the path point to version 7 of the php.exe I got from wamp I got the same error.

**Comment by leonieponsioen** [comment]
I had the same error, so I removed the executablePath setting and did the following steps:
Windows search for "environment"
click on edit environment variables (for your acount)
click on the "path" variable and press "Edit..."
click on "new"
write down the path to your php 7 files, for example: D:\php7 

(in your case it should be: C:\OpenServer\modules\php\PHP-7-x64 )

When starting visual studio code the error should be gone.

**Comment by dkimery256** [comment]
I did change the environment variable path. Still getting the error.

**Comment by mousetraps** [comment]
There are multiple failure-modes here, and they all have similar messages, so let's try and to be specific about which one we're talking about. 

| # | Error message | Source | Resolution | 
|---|-----------------|----------|-------------|
| 1 | `Cannot validate the php file. The php program was not found. Use the 'php.validate.executablePath' setting to configure the location of 'php'` | [built-in PHP validation provider](https://github.com/Microsoft/vscode/blob/c67ef57cda90b5f28499646f7cc94e8dcc5b0586/extensions/php/src/features/validationProvider.ts#L229) | Set `php.validate.executablePath` in FIle > Preferences > User Settings > Settings.json or add php to your PATH, which we'll fall back to if `php.validate.executablePath` is not set |
| 2 | `Failed to run php using path: {0}. Reason is unknown.` | [built-in PHP validation provider](https://github.com/Microsoft/vscode/blob/c67ef57cda90b5f28499646f7cc94e8dcc5b0586/extensions/php/src/features/validationProvider.ts#L231) | ??? - try running `My\php\validate\executablePath\php.exe -l -n -d log_errors=Off -f myFile.php`, and paste the output | 
| 3 | <ol><li>`PHP executable not found. You need PHP 7 installed and in your PATH`</li><li>`The language server needs at least PHP 7 installed and in your PATH. Version found: ' + version`</li><li>`Error spawning PHP: ' + err.message`</li><li>`Error parsing PHP version. Please check the output of php --version`</li></ol> | [vscode-php-intellisense extension](https://github.com/felixfbecker/vscode-php-intellisense/blob/1ac5a8fce3bd7adc0ca1a6a4e890777ca49c8ea5/src/extension.ts#L17) | Add the path to PHP7 to your environment variables, run `where php` and `php --version` to verify. You'll also need to relaunch VSCode (and the cmd line you are launching it from if you changed the environment variables) | 

So @FAST-JE, it looks like you ran into *#*1 first, you resolved it by adding `php.validate.executablePath`, and then ran into *#*3.1, so it would be great if you could verify that adding PHP7 to your env. variables resolves the issue. There's also an issue open to make the intellisense engine respect `php.validate.executablePath` https://github.com/felixfbecker/vscode-php-intellisense/issues/11, cc @felixfbecker

@dkimery256 I have no idea what error you're running into, but hopefully the table above will help narrow it down :smiley:

**Comment by mousetraps** [comment]
@dkimery256 oh, and one more thing! Just a hunch... you mentioned you changed the PATH from PHP5 to PHP7, which implies that this is not a new environment config, which implies that you've likely customized your PHP5 ini file, which implies that your PHP7 ini file might have different settings than your PHP5 ini file, which implies that something about those settings might have an effect on whether php commands execute as expected in VSCode or the php intellisense extension, which implies that WAMP has different default settings than default PHP that somehow impact basic execution, which implies that there's a good chance that it's related to some virtual filepath config, as is often the case with "cooked" environment setups.

So in summary, if you get errors after running `php --version` or `php -l -n -d log_errors=Off -f myFile.php`, also check your php.ini file, and, in particular, look out for potentially invalid filepaths.

**Comment by felixfbecker** [comment]
The message is indeed caused by my extension, thanks for clearing that up to detailed @mousetraps.

I'm haven't decided yet what settings "namespace" I should use for PHP IntelliSense. My thought was that the built-in PHP support hopefully soon gets removed from core? Because the validation and completion is conflicting with PHP IntelliSense and causing duplicate entries. Which is one of the reason why I thought it might be a good idea to use a generic `php.executablePath` setting, instead of `php.intellisense.executablePath`. Would be nice to hear some opinions on this from VS Code devs.

**Comment by FAST-JE** [comment]
@mousetraps I Added the path to PHP7 to your environment variables [screenshot](https://4.downloader.disk.yandex.ru/disk/38472c4a2bfc5d8f2426b993570ce9589bd41420e3086099a660b254fd52dfd3/57fb9c0d/y3LJERYif3pxLkOeQTlN_zxorrMOdF_KNjjGGzZ_DzAqV1br1cNUTQIxDYzvw11Dg_w9P510Jpu64Tjr84o_Cg%3D%3D?uid=0&filename=2016-10-10_12-46-31.png&disposition=inline&hash=&limit=0&content_type=image%2Fpng&fsize=1274&hid=49c760f8b2848a719f2c90a8ac6e0ec5&media_type=image&tknv=v2&etag=f6dd1fac54b8b2fd77fd4b4afbd41656)
and [screenshot settings.json](https://3.downloader.disk.yandex.ru/disk/7859a054a37920a10204c6935daba50d47dc1e96d93410b93335f242e5414a9c/57fb9ceb/y3LJERYif3pxLkOeQTlN_0cmWHHzgsjvNYV1JhnVLQ8HAZXZQP00QHxF5NnviIS0LD-xsf5z4Ls-CmCBoTLPFw%3D%3D?uid=0&filename=2016-10-10_12-50-01.png&disposition=inline&hash=&limit=0&content_type=image%2Fpng&fsize=8332&hid=13a2744adc996e722e713fa3d4884376&media_type=image&tknv=v2&etag=4f9772c58da550f552027dcc276ea799)
The problem is not solved.

**Comment by felixfbecker** [comment]
@FAST-JE You need to add the path of the directory, not the path to the exe. If you still experience problems please open an issue at my repository or chat with me on Gitter so we don't spam the VS Code dev folks here.

**Comment by mousetraps** [comment]
> I'm haven't decided yet what settings "namespace" I should use for PHP IntelliSense. My thought was that the built-in PHP support hopefully soon gets removed from core? Because the validation and completion is conflicting with PHP IntelliSense and causing duplicate entries. Which is one of the reason why I thought it might be a good idea to use a generic php.executablePath setting, instead of php.intellisense.executablePath. Would be nice to hear some opinions on this from VS Code devs.

@felixfbecker I like the idea of using `php.executablePath` rather than `php.validate.executablePath` both for the sake of simplification and because that appears to be how the broader ecosystem has tended to evolve (this is what both the Go and Python extensions do).

Also agreed re: removing built-in PHP support, because it appears to be the only language VSCode includes built-in completions for that is not one of the languages that VSCode is actually written in. No guarantees how soon that'll be, but I'll look into the potential implications this week. In the meantime, is there anything you can do to filter out duplicates in the completion list?

**Comment by mousetraps** [comment]
Closing this thread, as the primary issue has been resolved. @felixfbecker let's chat further about the executablePath stuff on a separate thread. One of the challenges will be figuring out how to resolve discrepancies with different versions of PHP between different extensions. Would be curious to hear your thoughts.

**Comment by BB-Bibo** [comment]
"php.executablePath":"C:\\wamp64\\bin\\php\\php7.0.10\\php.exe",

**Comment by dkimery256** [comment]
I got this working some time ago, forgot about the thread, sorry.  My problem was that I had my path set right in VS-Code, but my system var path I had included the php.exe in the path and that was causing issues.  Once I pointed my system var to the directory alone and not the exe, everything worked just fine.  Thanks for all the suggestions.

**Comment by devtekve** [comment]
For anyone having this issue, it is due the PATH, you must change your environment PATH. 
on Linux: nano ~/.bash_profile
On Windows: Edit environment Variables -> System Variables -> Path. Change the priority
![vscodeerror](https://cloud.githubusercontent.com/assets/7696966/26662864/f10dd390-4654-11e7-9d12-be0672064a12.png)

**Comment by Jhollman** [comment]
**SOLVED:**
Added in Settings:
```
{
    "php.executablePath": "C:\\PHP_7\\php.exe",
    "php.validate.executablePath": "C:\\PHP_7\\php.exe",
    ...
}
```

**Comment by okanime** [comment]
For Mac users:
This is what worked for me.
 {
    "php.executablePath": "/absolute path to your php 7 installation"
}

**Comment by xandydalan** [comment]
You installed PHP IntelliSense extension, and this error because of it.
So if you want to fix this problem go to this menu:
File -> Presences -> User Settings
Now you can see 2 window. In the right window add below codes:

{
    "php.validate.executablePath": "C:\\wamp64\\bin\\php\\php7.0.4\\php.exe",
    "php.executablePath": "C:\\wamp64\\bin\\php\\php7.0.4\\php.exe"
}

NOTICE: This address C:\\wamp64\\bin\\php\\php7.0.4\\php.exe is my php7.exe file address. Replace this address with own php7.exe.

Font: https://stackoverflow.com/questions/40287927/php-executable-not-found-install-php-7-and-add-it-to-your-path-or-set-the-php-e/45784662#45784662

**Comment by uurtech** [comment]
i've added php path to Windows Path then error disappears.

**Comment by Firegarden** [comment]
When I added the PHP execuable to the Path variable in windows it fixed the error for me.

**Comment by jens1o** [comment]
@Firegarden Can you ensure when you open a command prompt, that you can type `php -v` and get your current php version?

**Comment by abrandao** [comment]
Wy not just add a 'dont show option' for this useless function?
